### PR TITLE
Initial check-in of how-to for extracting article content from HTML using boilerpipe

### PR DESCRIPTION
**Pull request by rlvoyer**

### CODE REVIEW

File: extracting_article_content_from_HTML.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
+# Extract article content from HTML with Boilerpipe
+# Be sure to install boilerpipe (pip install boilerpipe)
```
Comment (srikris) [inline extracting_article_content_from_HTML.py]: Can you explain in the comment above what the input and output of the script is. Its not clear to me.

File: extracting_article_content_from_HTML.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+            extractor = Extractor(extractor='ArticleExtractor', html=src_content)
+            return extractor.getText()
+        except Exception as e:
+            logging.getLogger().error("Exception (%s) extracting from (%s)" % \
+                                      (e.message, src_content[0:25]))
+
+# Get the sample source data
+source_s3_path = "https://s3.amazonaws.com/dato-datasets/how-to/html_source_sample.zip"
+subprocess.check_call(["wget", source_s3_path])
+subprocess.check_call(["unzip", "html_source_sample.zip"])
```
Comment (srikris) [inline extracting_article_content_from_HTML.py]: Use zipfile to extract the zip.

File: extracting_article_content_from_HTML.py [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    if src_content and src_content.strip():
+        try:
+            extractor = Extractor(extractor='ArticleExtractor', html=src_content)
+            return extractor.getText()
+        except Exception as e:
+            logging.getLogger().error("Exception (%s) extracting from (%s)" % \
+                                      (e.message, src_content[0:25]))
+
+# Get the sample source data
+source_s3_path = "https://s3.amazonaws.com/dato-datasets/how-to/html_source_sample.zip"
```
Comment (srikris) [inline extracting_article_content_from_HTML.py]: This won't work if they don't have wget. Mac doesn't come with wget installed.

File: extracting_article_content_from_HTML.py [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+dataset = gl.SFrame({"filename": source_sa,
+                     "source": source_sa.apply(
+                         lambda filename: codecs.open(
+                             os.path.join(source_dir, filename), "r", "utf8").read())})
+
+# Extract clean text for each HTML file and create new column caled "content"
+dataset["content"] = dataset["source"].apply(
+    lambda src: (extract_article_content(src) or None) if src else None)
+
+dataset
```
Comment (srikris) [inline extracting_article_content_from_HTML.py]: This is not needed. You can put a comment in the end about what the output looks like? That might be easier.

File: extracting_article_content_from_HTML.py [hunk 5]
```diff
@@ -0,0 +1,40 @@
+# Extract article content from HTML with Boilerpipe
```
Comment (srikris) [inline extracting_article_content_from_HTML.py]: Add to the README.md

File: extracting_article_content_from_HTML.py (hunk 1 above)
Comment (rlvoyer) [inline extracting_article_content_from_HTML.py]: Input, and output? The script, like most of our HOW-TOs is fully self-contained. It downloads some source HTML files from S3 and then extracts the textual content into an SFrame. Not quite clear what you're looking for. A description of the boilerpipe module? How pip works?

**Pull request merged**